### PR TITLE
[0.2] Upgrade to lair keystore 0.3.0

### PR DESCRIPTION
**Pull request by ThetaSinner**
### Summary

This now also includes a serde pin and a dependencies lock update

### TODO:
- [ ] CHANGELOG(s) updated with appropriate info
- [ ] Just before pressing the merge button, ensure new entries to CHANGELOG(s) are still under the _UNRELEASED_ heading

### CODE REVIEW

File: crates/kitsune_p2p/kitsune_p2p/Cargo.toml [hunk 1]
```diff
@@ -43,7 +43,7 @@ thiserror = "1.0.22"
 tokio = { version = "1.27", features = ["full"] }
 tracing = "0.1"
 tokio-stream = "0.1"
-tx5 = { version = "=0.0.1-alpha.18", optional = true }
+tx5 = { version = "=0.0.2-alpha", optional = true }
```
Comment (neonphog) [inline crates/kitsune_p2p/kitsune_p2p/Cargo.toml:46]: per https://github.com/holochain/holochain/pull/2669#discussion_r1296006148 please update the other tx5 refs as well.

**Review by neonphog — APPROVED** [review]

**Comment by steveej** [comment]
we're seeing this error

       > 95 |         D: serde::Serialize + std::fmt::Debug,
       >    |            ^^^^^^^^^^^^^^^^ required by this bound in `holochain_integrity_types::VerifySignature::new`
       > help: consider further restricting this bound
       >    |
       > 15 |     D: serde::Serialize + std::fmt::Debug + holochain_wasmer_guest::Serialize,
       >    |                                           +++++++++++++++++++++++++++++++++++
       >
       > For more information about this error, try `rustc --explain E0277`.
       > error: could not compile `hdi` due to previous error

my understanding is that `hdi::serde::Serialize` and `holochain_wasmer_guest::Serialize` originate from different versions of `serde`. i'm currently testing if the round-trip of synchronizing the Cargo.lock files (`nix develop .#coreDev --command script-cargo-regen-lockfiles`) helps here. if it does i'll push the commit to this branch.

**Comment by ThetaSinner** [comment]
It's not that simple, please don't push that

**Comment by ThetaSinner** [comment]
I've already fixed this same issue for the 0.1 branch

**Review by steveej — APPROVED** [review]
thank you!

**Pull request merged**